rust/kernel/types.rs

```diff
@@ -46,6 +46,18 @@ pub trait ForeignOwnable: Sized {
     /// Additionally, all instances (if any) of values returned by [`ForeignOwnable::borrow`] for
     /// this object must have been dropped.
     unsafe fn from_foreign(ptr: *const core::ffi::c_void) -> Self;
+
+    /// Converts a foreign-owned object back to a Rust-owned one.
+    ///
+    /// similar to `from_foreign`, but if `ptr` is null, it will return `None`.
```

Comment:
Perhaps:

```suggestion
    /// Tries to convert a foreign-owned object back to a Rust-owned one.
    ///
    /// Like [`from_foreign`], but returns [`None`] if `ptr` is null.
```

(I didn't check if the intra-doc links would work as-is)